unic-langid-impl/src/lib.rs

```diff
@@ -94,7 +111,7 @@ impl LanguageIdentifier {
     ///
     /// assert_eq!(li.to_string(), "fr-CA");
     /// ```
-    pub fn from_parts<S: AsRef<str>>(
+    pub fn from_parts<S: AsRef<[u8]>>(
```

Comment:
Hmm, this means that using `from_parts(None, None, None, &[])` would not compile and you'd need to specify the type in at least one, but I guess as long as there's a `::default()` that does it you're good.

Reply:
yeah, I don't think you'd ever want to `from_parts` with just nones.